client/zone/js-modules/map-color-finder.js

```diff
@@ -0,0 +1,189 @@
+export class MapColorFinder {
```

Comment:
Maybe `FourColorTheoremSolver` is a better class name because it actually describe what it is?
At least this should be noted in the comment that this is a implementation of the theorem.

Reply:
Agreed. The name is far too generic.